src/dsp_tools/analyse_xml_data/extract_links_from_XML.py

```diff
@@ -6,62 +6,65 @@
 
 
-def _create_info_from_xml_for_graph(root: etree._Element) -> tuple[list[ResptrLink], list[XMLLink], set[str]]:
+def _create_info_from_xml_for_graph(
+    root: etree._Element,
+) -> tuple[etree._Element, list[ResptrLink], list[XMLLink], list[str]]:
     """Create instances of the classes ResptrLink and XMLLink from the root of the XML file."""
     resptr_instances = []
     xml_instances = []
-    all_resource_ids = set()
+    all_resource_ids = []
     for resource in root.iter(tag="{https://dasch.swiss/schema}resource"):
         resptr, xml, subject_id = _create_info_from_xml_for_graph_from_one_resource(resource)
-        if resptr:
-            resptr_instances.extend(resptr)
-        if xml:
-            xml_instances.extend(xml)
-        if subject_id:
-            all_resource_ids.add(subject_id)
-    return resptr_instances, xml_instances, set(all_resource_ids)
+        all_resource_ids.append(subject_id)
+        resptr_instances.extend(resptr)
+        xml_instances.extend(xml)
+    return root, resptr_instances, xml_instances, all_resource_ids
 
 
 def _create_info_from_xml_for_graph_from_one_resource(
     resource: etree._Element,
 ) -> tuple[list[ResptrLink], list[XMLLink], str]:
     subject_id = resource.attrib["id"]
-    resptr_links, xml_links = _get_all_links_from_one_resource(resource)
-    resptr_link_objects = []
-    xml_link_objects = []
-    if resptr_links:
-        resptr_link_objects = [ResptrLink(subject_id, object_id) for object_id in resptr_links]
-    if xml_links:
-        xml_link_objects = [XMLLink(subject_id, x) for x in xml_links]
-    return resptr_link_objects, xml_link_objects, subject_id
-
-
-def _get_all_links_from_one_resource(resource: etree._Element) -> tuple[list[str], list[set[str]]]:
-    resptr_links: list[str] = []
-    xml_links: list[set[str]] = []
+    resptr_links, xml_links = _get_all_links_from_one_resource(subject_id, resource)
+    return resptr_links, xml_links, subject_id
+
+
+def _get_all_links_from_one_resource(
+    subject_id: str, resource: etree._Element
+) -> tuple[list[ResptrLink], list[XMLLink]]:
+    resptr_links: list[ResptrLink] = []
+    xml_links: list[XMLLink] = []
     for prop in resource.getchildren():
         match prop.tag:
             case "{https://dasch.swiss/schema}resptr-prop":
-                resptr_links.extend(_extract_ids_from_one_resptr_prop(prop))
+                resptr_links.extend(_create_class_instance_resptr_link(subject_id, prop))
             case "{https://dasch.swiss/schema}text-prop":
-                xml_links.extend(_extract_ids_from_text_prop(prop))
+                xml_links.extend(_create_class_instance_text_prop(subject_id, prop))
     return resptr_links, xml_links
 
 
-def _extract_ids_from_one_resptr_prop(resptr_prop: etree._Element) -> list[str]:
-    return [x.text for x in resptr_prop.getchildren() if x.text]
+def _create_class_instance_resptr_link(subject_id: str, resptr_prop: etree._Element) -> list[ResptrLink]:
```

Comment:
not a fan of this function name. maybe `_create_resptr_link` or `_create_resptr_link_object`?